collections-contrib/src/main/scala/strawman/collection/decorators/HasImmutableMapOps.scala

```diff
@@ -0,0 +1,43 @@
+package strawman.collection
+package decorators
+
+/**
+  * Type class witnessing that a collection type `C` has
+  * a conversion to `immutable.MapOps[K, V, _, _]`.
+  *
+  * @see [[scala.collection.decorators.HasIterableOps]]
+  */
+trait HasImmutableMapOps[C] extends HasMapOps[C] {
+
+  // Convenient intermediate type definitions to satisfy type bounds
+  protected type _CC[X, +Y] <: immutable.MapOps[X, Y, _CC, _]
+  protected type _C <: immutable.MapOps[K, V, _CC, _C]
```

Comment:
Again, if the user needs to interact with this it should have a better name, e.g. `MapType`.